### PR TITLE
VZ-5478: Upgrade of all previous supported releases to the next minor release

### PR DESCRIPTION
**Pull request by Abhijithang**
Automated tests for upgrade of all previous supported releases to the next minor release

Fixes VZ-5478

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: ci/upgrade-paths/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,879 @@
+// Copyright (c) 2021, 2022, Oracle and/or its affiliates.
```
Comment (kminder) [inline ci/upgrade-paths/Jenkinsfile:1]: Doesn't need 2021

File: ci/upgrade-paths/JenkinsfileTestTrigger [hunk 2]
```diff
@@ -0,0 +1,439 @@
+// Copyright (c) 2020, 2022, Oracle and/or its affiliates.
```
Comment (kminder) [inline ci/upgrade-paths/JenkinsfileTestTrigger:1]: Just 2022?

File: ci/upgrade-paths/JenkinsfileTestTrigger [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+                    withCredentials([file(credentialsId: 'jenkins-to-slack-users', variable: 'JENKINS_TO_SLACK_JSON')]) {
+                        def userMappings = readJSON file: JENKINS_TO_SLACK_JSON
+                        SUSPECT_LIST = getSuspectList(commitList, userMappings)
+                        echo "Suspect list: ${SUSPECT_LIST}"
+                    }
+                }
+            }
+        }
+
+        stage ('Kick off parallel tests') {
```
Comment (kminder) [inline ci/upgrade-paths/JenkinsfileTestTrigger:140]: Seems like this could be named better.

File: ci/upgrade-paths/JenkinsfileTestTrigger [hunk 4]
```diff
@@ -0,0 +1,439 @@
...
+            parallel {
+                stage('Upgrade every release to Release Candidate') {
+                steps {
+                    script {
+                           sh """
+                                cd ${workspace}
+                                ls -ltr
```
Comment (kminder) [inline ci/upgrade-paths/JenkinsfileTestTrigger:147]: What is the ls for?

File: ci/upgrade-paths/JenkinsfileTestTrigger (hunk 2 above)
Comment (Abhijithang) [reply ci/upgrade-paths/JenkinsfileTestTrigger:1]: Do I need to run the attribution helper for this?

File: ci/upgrade-paths/JenkinsfileTestTrigger [hunk 5]
```diff
@@ -0,0 +1,439 @@
...
+        string (name: 'VERRAZZANO_OPERATOR_IMAGE',
+                        defaultValue: 'NONE',
+                        description: 'This is for manually testing only where someone needs to use a specific operator image, otherwise the default value of NONE is used',
+                        trim: true)
+        string (name: 'WILDCARD_DNS_DOMAIN',
+                        defaultValue: 'nip.io',
+                        description: 'This is the wildcard DNS domain',
+                        trim: true)
+        booleanParam (description: 'Whether to emit metrics from the pipeline', name: 'EMIT_METRICS', defaultValue: true)
+        string (name: 'TAGGED_TESTS',
```
[the remark this answers is not in the document]
Comment (Abhijithang) [reply ci/upgrade-paths/JenkinsfileTestTrigger:51]: Yes they are used in ginkgo tests.

File: ci/upgrade-paths/JenkinsfileTestTrigger (hunk 4 above)
Comment (Abhijithang) [reply ci/upgrade-paths/JenkinsfileTestTrigger:147]: ls -ltr, isn't required. cd ${workspace} is to make sure I am in the verrazzano repo directory to read the git release tags.

File: ci/upgrade-paths/JenkinsfileTestTrigger (hunk 3 above)
Comment (Abhijithang) [reply ci/upgrade-paths/JenkinsfileTestTrigger:140]: "Compute Upgrade Jobs and Trigger Parallel Tests" Does this sound good?

**Review by kminder — APPROVED** [review]
Approved.

**Pull request merged**